code/controllers/subsystem/ships.dm

```diff
@@ -164,48 +164,52 @@ SUBSYSTEM_DEF(ship)
 	var/datum/ship_attack/attack_data = W.attack_data
 
-	if(prob(player_evasion_chance))
+	if(prob(player_evasion_chance)) //Chance to miss
 		broadcast_message("<span class=notice> Enemy ship ([S.name]) fired their [W.name] but missed!</span>",success_sound,S)
-	else
-		if(SSstarmap.ftl_shieldgen && SSstarmap.ftl_shieldgen.is_active() && !W.attack_data.shield_bust)
-			SSstarmap.ftl_shieldgen.take_hit()
+		return FALSE
+
+	if(~W.attack_data.unique_effect & SHIELD_PENETRATE && SSstarmap.ftl_shieldgen && SSstarmap.ftl_shieldgen.is_active()) //If I penetrate shields I don't give one fucking shit about your shields so dont even check it, else check if we have a shield
+		if(W.attack_data.shield_damage) //If I do shield damage, fuck those shields up.
+			SSstarmap.ftl_shieldgen.take_hit(W.attack_data.shield_damage)
 			broadcast_message("<span class=warning>Enemy ship ([S.name]) fired their [W.name] and hit! Hit absorbed by shields.",error_sound,S)
-			for(var/area/shuttle/ftl/A in world)
-				A << shield_hit_sound
-		else
-			var/obj/docking_port/mobile/D = SSshuttle.getShuttle("ftl")
-
-			if(W.attack_data.shield_bust)
-				SSstarmap.ftl_shieldgen.take_hit()
-
-			var/list/target_list = D.return_unordered_turfs()
-			var/turf/target
-			while(!target)
-				var/turf/T = pick(target_list)
-				if(!istype(T,/turf/open/space))
-					target = T
-			if(attack_data.unique_effect == FRAGMENTED_SHOT) //If our shot hits multiple times
-				/var/turf/target_sub
-				new /obj/effect/temp_visual/ship_target(target, attack_data) //Initial hit
-				for(var/I = 1 to attack_data.unique_effect_modifier_one) //Loop for each fragment
-					spawn(attack_data.warning_time+I)//Saves spamming many audio queues at once
-						target_sub = locate(target.x + rand(-attack_data.unique_effect_modifier_two,attack_data.unique_effect_modifier_two),target.y + rand(-attack_data.unique_effect_modifier_two,attack_data.unique_effect_modifier_two), target.z)
-						new /obj/effect/temp_visual/ship_target(target_sub, attack_data)
-			else //Normal single shot
-				new /obj/effect/temp_visual/ship_target(target, attack_data) //thingy that handles the ship projectile
-
-			spawn(50)
-
-				if(W.attack_data.shield_bust)
-					broadcast_message("<span class=warning>Enemy ship ([S.name]) fired their [W.name], which pierced the shield and hit! Hit location: [target.loc].</span>",error_sound,S) //so the message doesn't get there early
-				else
-					broadcast_message("<span class=warning>Enemy ship ([S.name]) fired their [W.name] and hit! Hit location: [target.loc].</span>",error_sound,S) //so the message doesn't get there early
-				for(var/mob/living/carbon/human/M in GLOB.player_list)
-					if(!istype(M.loc.loc, /area/shuttle/ftl))
-						continue
-					var/dist = get_dist(M.loc, target.loc)
-					shake_camera(M, dist > 20 ? 3 : 5, dist > 20 ? 1 : 3)
+			SSstarmap.ftl_sound('sound/weapons/ship_hit_shields.ogg')
```

Comment:
shield_hit_sound

Reply:
oh woops i forgot to upload the new sound I had

Reply:
wait no u r a liar